### ISSUE TITLE
does grabbing urls from chrome work?  I'm not seeing anything.

### ISSUE THREAD

**Issue by brizzbane** [ActivityWatch/activitywatch#162]
It's great at grabbing titles of everything, but I do not see urls anywhere.  I'm just running default--everything that the docs say to get going.

**Comment by johan-bjareholt** [comment]
If you install the browser extension it will collect the data, but the visualization is still in progress and will be released in the next version. It is almost done and beta will hopefully be available soon if you want to try it out. The only visualization of the browser activity implemented is a summary of which websites you spend the most time on, we have plans for more ways to visualize it in later versions.

So I would recommend to have the extension running and once the beta releases within a couple of weeks you can simply install that and you'll see the data collected during that period.

**Comment by johan-bjareholt** [comment]
> Anyways--with this extension, an idea, allow you to specify a host to send data to.

Yeah, using a remote aw-server for an aw-client has been brought up before.
https://github.com/ActivityWatch/activitywatch/issues/142#issuecomment-346936710
TL;DR You can configure it to work but it is insecure.

The main issue with chromebooks which makes it impossible however is that you cannot run aw-watcher-afk on it, so if we would make the webextension send data to a remote aw-server it would still not be aware if you are at the computer or not. Even if you were able to install it, chromeOS runs wayland which we do not support yet (it's currently impossible to implement due to the compositor missing the ability to expose mouse movement, keyboard presses and window status to other programs).

**Comment by brizzbane** [comment]
Hey, I was thinking about this, and had a question for you.

Have you thought about a solution that does not use any of the aw-python programs at all?  (for Chromebooks, or Chrome browser in general)  I.E.  The extension itself would do all of the monitoring of activity/afk check, etc.  I have not really worked with extensions much, but my brother showed me something that he built.  I know that if you give Chrome the permissions, you get a *ton* of access.

I know that this would be a pretty big divergence from how you have things set up now...  and would be a project in itself.  I had just wondered if you had considered that?  (and if there are obvious problems with going that route)--or if you hadn't, and this might be a possibility?

I know it would be a lot of work just for 'chromebooks', but IMHO, people might really like it (outside of chromebooks), just because of the syncing nature of extensions in Chrome--and this would be a 'lightweight' solution for computers they may use that they don't want to take the time to install... whatever..  (Just trying to maybe present a case for it! :D)

**Comment by johan-bjareholt** [comment]
>  I know that if you give Chrome the permissions, you get a ton of access. 

This is the issue why it is technically impossible. Chrome gives you a lot of permissions to the webbrowser yes, but it does not give you any permission at all to the OS, which is needed to know whether the user is AFK or not (detecting mouse movement/keyboard presses).

We could also just assume that if the person is using a chromebook he/she is always at the computer and always has the chrome window focused, but at that point it I would guess the actual usage times presented in activitywatch would be off by too much to even be useful.

**Comment by ErikBjare** [comment]
As @johan-bjareholt said, it seems like a lot of work for something that wouldn't even end up being that good. Maybe someday, but not now.

**Comment by ErikBjare** [comment]
Closing, thanks for the feedback @brizzbane! 👍

**Comment by brizzbane** [comment]
Can't you use JavaScript to detect mouse movement/keyboard presses?

**Comment by ErikBjare** [comment]
Not on the OS level, although there seems to be an idle API: https://groups.google.com/a/chromium.org/forum/m/#!topic/chromium-extensions/EWu2xPS2CK4

Maybe that could work, but since we can't run the server locally it would still be too much work to implement properly.

**Comment by marcusquinn** [comment]
Don't know how they do it but TimeCamp definitely works on ChromeOS - although would be happy to switch to ActivityWatch if you figure it out:

https://www.timecamp.com
https://chrome.google.com/webstore/detail/timecamp/ohbkdjmhoegleofcohdjagmcnkimfdaa?hl=en
